### PR TITLE
Add IT tests to test installation according to docs

### PR DESCRIPTION
**Pull request by dliappis**
As a followup to #1185 this commit adds a few IT tests to validate the
Rally installation steps in the docs.

### CODE REVIEW

**Comment by danielmitterdorfer** [comment]
The CI build has failed with:

```

13:37:54 pyenv: version `3.8.7' is not installed (set by /var/lib/jenkins/workspace/elastic+rally+pull-request/.python-version)
13:37:54 Created python3 venv under /var/lib/jenkins/workspace/elastic+rally+pull-request/.venv.
13:37:54 . .venv/bin/activate; python3 -m pip install --upgrade pip setuptools wheel
13:37:54 /bin/bash: .venv/bin/activate: No such file or directory
13:37:54 pyenv: version `3.8.7' is not installed (set by /var/lib/jenkins/workspace/elastic+rally+pull-request/.python-version)
13:37:54 Makefile:65: recipe for target 'install-user' failed
```

I think we need to add:

```
pyenv install --skip-existing $(PYVER4DOCS)
```

in the `prereq` target of the Makefile.

**Comment by dliappis** [comment]
Hm 

> The CI build has failed with:

Hm I think something else is wrong as actually I don't have 3.8.7 in my local pyenv's and still it's working fine. Will investigate

```
$ pyenv versions
  system
  2.7.18
  2.7.18/envs/ve
* 3.8.0 (set by /home/dl/source/elastic/rally/.python-version)
  3.8.5
* 3.9.0 (set by /home/dl/source/elastic/rally/.python-version)
  ve
```

**Comment by dliappis** [comment]
I pushed https://github.com/elastic/rally/pull/1193/commits/e0081981c8996450afa39b8274f82151288c5272 so that standard `make prereq` doesn't install a Python version that isn't needed locally (we use a Docker image for the specific IT test).

**Comment by dliappis** [comment]
@danielmitterdorfer I refactored as per our offline discussion in https://github.com/elastic/rally/pull/1193/commits/3374548ee03b3fee8533d1ba92c6e373e5e9416f to not introduce a new `DOCS` CI variable but instead rely on the existing ones. While at it, bumped the patch versions. Feel free to review after CI gets green.

**Review by danielmitterdorfer — COMMENTED** [review]
Thanks! I left a couple of suggestions.

File: Makefile [hunk 1]
```diff
@@ -21,8 +21,8 @@ PYENV_REGEX = .pyenv/shims
 PY_BIN = python3
 # https://github.com/pypa/pip/issues/5599
 PIP_WRAPPER = $(PY_BIN) -m pip
-PY38 = $(shell jq '.python_versions.PY38' .ci/variables.json)
-PY39 = $(shell jq '.python_versions.PY39' .ci/variables.json)
+export MIN_PY_VER = $(shell jq -r '.python_versions.PY38' .ci/variables.json)
```
[the remark this answers is not in the document]
Comment (dliappis) [reply Makefile:24]: We can; what is the purpose is we won't be using `PY38` though?

File: it/installation_test.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+These tests ensure the validatity of Rally installation instructions (as shown in docs)
+"""
+
+import os
+import warnings
+
+import it
+
+_CI_VARS = ".ci/variables.json"
+_FALLBACK_MIN_PY_VER = "3.8.7"
```
[the remark this answers is not in the document]
Comment (dliappis) [reply it/installation_test.py:28]: Sure, this is what we did before this refactoring in https://github.com/elastic/rally/pull/1193/commits/3374548ee03b3fee8533d1ba92c6e373e5e9416f#diff-c0d0ee48d5c24db80722a8fef4cb274fea04db4fb2cd8c4147697758de4e9192L31-L38, will bring this back.

File: it/__init__.py [hunk 3]
```diff
@@ -90,6 +90,23 @@ def race(cfg, command_line):
...
+    Executes a given command_line in a subshell.
+
+    :param command_line: (str) The command to execute
+    :return: (int) the exit code
+    """
+
+    return os.system(command_line)
+
+
+def command_in_docker(command_line, python_version="3.8.7"):
```
[the remark this answers is not in the document]
Comment (dliappis) [reply it/__init__.py:104]: Good idea.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply Makefile:24]: The idea behind this is that we derive the version variables always consistently. We should use the variable `PY38` though when installing Python versions in `make install`. `MIN_PY_VER` can be used anywhere where we don't need to refer to a specific version but we want to convey the intent that we're looking for the minimum supported one.

**Comment by dliappis** [comment]
CI failed specifically with `py39` (`test_tar_distributions[es-it]`) and since nothing has changed it, I will just wait for the new CI results triggered by https://github.com/elastic/rally/pull/1193/commits/ae8ae56573e71f6c17660f287fd317b018637e8e and consider the failure transient.

**Comment by dliappis** [comment]
Hm another CI failure this time with 

```
************* Module tests.driver.scheduler_test
14:24:42 tests/driver/scheduler_test.py:95:28: E1102: s.next is not callable (not-callable)
```

but it's not reproducible locally with exactly the same `pylint` version.

@elasticmachine please test this

**Comment by dliappis** [comment]
@elasticmachine test this please

**Comment by dliappis** [comment]
@danielmitterdorfer CI is green (at last!) could you take another pass when you have time?

**Review by danielmitterdorfer — APPROVED** [review]
Great that we have a test for this now. Thanks for iterating. LGTM

**Comment by dliappis** [comment]
@danielmitterdorfer thanks a lot for the review!

**Pull request merged**